erc20-migration/hardhat.config.js

```diff
@@ -230,22 +253,172 @@ task("restoreEON", "Restores EON accounts", async (taskArgs, hre) => {
   console.log("\n\n***************************************************************");
   console.log("                 Checking distributed tokens");
   console.log("***************************************************************\n");
-  let num_of_mismatches = 0;
+  let mismatch_messages = [];
+  let count = 1;
   for (const [address, balance] of accounts) {
+    const progressPercentage = Math.floor(count / accounts.length * 100);
+    process.stdout.clearLine();
+    process.stdout.cursorTo(0);
+    process.stdout.write(`Progress: ${progressPercentage}%`);
     let currentBalance = await ZENToken.balanceOf(address);
     if (currentBalance != balance) {
-      console.log("Balance of address " + address + " different from expected - expected: " + balance + ", actual: " + currentBalance);
-      num_of_mismatches++;
+      mismatch_messages.push("Balance of address " + address + " different from expected - expected: " + balance + ", actual: " + currentBalance);
     }
+    count++;
   }
 
   console.log("\n***************************************************************");
-  console.log("             Checking end. Mismatches found: " + num_of_mismatches);
+  console.log("             Checking end. Mismatches found: " + mismatch_messages.length);
+  if (mismatch_messages.length > 0) {
+    for (msg of mismatch_messages)
+      console.log(msg);
+  }
   console.log("***************************************************************");
 
 
 });
 
+task("restoreZEND", "Restores ZEND accounts", async (taskArgs, hre) => {
```

Comment:
aslo here let's separate the task in 2 parts:

- calculation of the hash
- execution
(In the migration day the first step will be executed by multiple developers, for double-check)

In the execution part we may add the hash as input parameter - to further check nothing changed in the middle

Reply:
Done